pennylane/templates/state_preparations/mottonen.py

```diff
@@ -76,47 +76,57 @@ def _matrix_M_entry(row, col):
 
 
 def _compute_theta(alpha):
-    """Calculates the rotation angles from the alpha vector.
+    """Maps the angles alpha of the multi-controlled rotations decomposition of a uniformly controlled rotation
+     to the rotation angles used in the Gray code implementation.
 
     Args:
         alpha (array[float]): alpha parameters
 
     Returns:
         (array[float]): rotation angles theta
     """
+    ln = alpha.shape[0]
     k = np.log2(alpha.shape[0])
-    factor = 2 ** (-k)
 
-    theta = sparse.dok_matrix(alpha.shape, dtype=np.float64)  # type: sparse.dok_matrix
+    M_trans = np.zeros(shape=(ln, ln))
+    for i in range(len(M_trans)):
+        for j in range(len(M_trans[0])):
+            M_trans[i, j] = _matrix_M_entry(j, i)
```

Comment:
I'm happy if there is a better way to fill a matrix from a function, or even to implement a matrix-vector product without storing the matrix. But couldn't make it work. This seems to be reasonably fast.